src/ansible_navigator/utils/functions.py

```diff
@@ -159,6 +160,57 @@ def clear_screen() -> None:
             print()
 
 
+class DictMergeError(Exception):
+    """Custom exception for a dict merge error."""
+
+
+Mergeable = Optional[
+    Union[
+        bool,
+        Dict,
+        List,
+        str,
+    ]
+]
+
+
+def dict_merge_in_place_list_replace(left: Mergeable, right: Mergeable):
+    """Merges right into left in place and returns merged result
+
+    If left is a primitive, right will be used
+    If left key is a list or tuple right will be used.
+    If left and right have a common key, they will be dict merged
+    If left is missing a key from right, the key from right will be added
+
+    :param left: Dict to be merged into
+    :param right: Dict to be merged from
+    :return: Merged dict
+    """
+    key = None
+    try:
+        if left is None or isinstance(left, (str, int, float, bool)):
+            # Border case for first run or if a is a primitive
```

Comment:
should this be

```suggestion
            # Border case for first run or if left is a primitive
```
?